src/baldr/location.cc

```diff
@@ -144,7 +144,10 @@ Location Location::FromRapidJson(const rapidjson::Value& d){
   Location location{{*lon,*lat}, stop_type};
 
   location.date_time_ = GetOptionalFromRapidJson<std::string>(d, "/date_time");
-  location.heading_ = GetOptionalFromRapidJson<float>(d, "/heading");
+
+  if (d.HasMember("heading"))
+    location.heading_ = d["heading"].IsFloat() ?  GetOptionalFromRapidJson<float>(d, "/heading").get_value_or(-1) : GetOptionalFromRapidJson<int>(d, "/heading").get_value_or(-1);
```

Comment:
you may as well just get `float` here too. and if its not `boost::none` then set `location.heading_`